lib/crystalball/map_generator/factory_bot_strategy/dsl_patch.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Crystalball
+  class MapGenerator
+    class FactoryBotStrategy
+      # Module to add new patched `factory` method to FactoryBot::Syntax::Default::DSL and FactoryBot::Syntax::Default::ModifyDSL
+      module DSLPatch
+        class << self
+          # Patches `FactoryBot::Syntax::Default::DSL#factory` and `FactoryBot::Syntax::Default::ModifyDSL#factory`.
+          def apply!
+            classes_to_patch.each { |klass| klass.prepend DSLPatch }
+          end
+
+          private
+
+          def classes_to_patch
+            [
+              FactoryBotStrategy.factory_bot_constant::Syntax::Default::DSL,
+              FactoryBotStrategy.factory_bot_constant::Syntax::Default::ModifyDSL
+            ]
+          end
+        end
+
+        # Will replace original `FactoryBot::Syntax::Default::DSL#factory` and `FactoryBot::Syntax::Default::ModifyDSL#factory`.
+        # Pushes path of a factory to `FactoryBotStrategy.factory_definitions` and calls original `factory`
+        def factory(*args, &block)
+          factory_path = cb_fetch_factory_path
+          name = args.first.to_s
```

Comment:
you can call for `super` first and then rely on object returned instead of rely on method interface which can be a subject to change by factory bot authors.

Reply:
It returns not a factory